nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/BaseRealm.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 1997-2013 Oracle and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
+ * or packager/legal/LICENSE.txt.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at packager/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * Oracle designates this particular file as subject to the "Classpath"
+ * exception as provided by Oracle in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+// Portions Copyright [2018-2019] Payara Foundation and/or affiliates
+package com.sun.enterprise.security.auth.realm;
+
+import static com.sun.enterprise.security.BaseRealm.JAAS_CONTEXT_PARAM;
+import static com.sun.enterprise.security.auth.realm.RealmsManagerStore._getRealmsManager;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * This class is the base for all Payara Realm classes, and contains common
+ * object state such as the realm name, general properties and the assign groups.
+ * 
+ * @author Arjan Tijms
+ *
+ */
+public abstract class BaseRealm extends AbstractRealm implements Comparable<Realm> {
```

Comment:
What happens with `com.sun.enterprise.security.BaseRealm`? We shouldn't remove it, as it may be in use by users, and having two classes with similar name is confusing.

Reply:
Good point, let's think of another name for that one then.

Reply:
After a chat settled on `AbstractStatefulRealm`. Thx again @pdudits